packages/connector-puppeteer/src/lib/authenticators.ts

```diff
@@ -0,0 +1,88 @@
+import * as puppeteer from 'puppeteer-core';
+import { ConnectorOptions } from '../connector';
+
+export type HTTPAuthConfig = {
+    user: string;
+    password: string;
+}
+
+export type AuthConfig = {
+    user: {
+        selector: string;
+        value: string;
+    };
+    next?: {
+        selector: string;
+    };
+    password: {
+        selector: string;
+        value: string;
+    };
+    submit: {
+        selector: string;
+    };
+};
+
+/**
+ * `connector-puppeteer` action to authenticate on a website using Basic HTTP Auth.
+ * To be used on `beforeTargetNavigation`.
+ */
+export const basicHTTPAuth = async (page: puppeteer.Page, config: ConnectorOptions) => {
+    if (!(config && config.auth &&
```

Comment:
This is a little hard to read given the wrapped negation. Perhaps breaking this into multiple `if (condition) { return }` blocks would be a bit easier to follow?

```ts
if (!config || !config.auth) {
    return;
}
if (typeof config.auth.user !== 'string' || typeof config.auth.password !== 'string') {
    return;
}
```

Reply:
Done